### PR TITLE
Fix script tag helper dependencies inheriting the script's HTML attributes

### LINKED ISSUE #15375: Script type="module" issue
### Describe the bug

When using `<script asp-name="something" type="module">` the dependent scripts also gets a type="module" parameter added to its script tag if it is defined in a ResourceManagementOptionsConfiguration as a dependency.

### To Reproduce

```
            _manifest
                .DefineScript("media")
                .SetUrl("~/OrchardCore.Media/Scripts/media2.js")
                .SetDependencies("vuejs:3", "sortable", "vuedraggable:3")
                .SetVersion("2.0.0");
```

`<script type="module" asp-name="media" version="2" at="Foot"></script>`

Look at the script added and all of the dependent scripts will have a type="module" added to them.

### Expected behavior

Only the main script should have type="module" added.

### PR DESCRIPTION
**Pull request by sarahelsaig**
Also some code cleanup in `ScriptTagHelper` to keep it DRY.

Fixes #15375

### CODE REVIEW

**Comment by Skrypt** [comment]
I don't remember exactly why we were inheriting these script tag properties. I hope it won't break anything else. Maybe we should have unit tests to make sure that we cover the specs that were needed ... if we ever find them.

**Comment by sarahelsaig** [comment]
To be fair this setting inheritance was set up [8 years ago](https://github.com/OrchardCMS/OrchardCore/commit/ac19c1425d16b817eb625b8fb498e18a544d3427#diff-2d2ac0e5caedcbecf0fd290ee642c98e3b9970746832eeef80d9a5832d31b10fR343-R350) in the same PR where the `Orchard.ResourceManagement` project was created. So it may just be a case of initial over-engineering.

**Comment by Skrypt** [comment]
Yeah and it was copied from Orchard 1 so add another 10 years on that. 😄

**Comment by sarahelsaig** [comment]
Yeah, I've looked into it and guess what?

image

File: src/OrchardCore/OrchardCore.ResourceManagement/TagHelpers/ScriptTagHelper.cs [hunk 1]
```diff
@@ -52,7 +52,10 @@ public override async Task ProcessAsync(TagHelperContext context, TagHelperOutpu
         {
             output.SuppressOutput();
 
-            if (string.IsNullOrEmpty(Name) && !string.IsNullOrEmpty(Src))
+            var hasName = !string.IsNullOrEmpty(Name);
+            var hasSource = !string.IsNullOrEmpty(Src);
+
+            if (!hasName && hasSource)
```
Comment (Skrypt) [inline src/OrchardCore/OrchardCore.ResourceManagement/TagHelpers/ScriptTagHelper.cs:58]: ok for readability.

**Review by Skrypt — APPROVED** [review]

**Pull request merged**